### PR TITLE
add lr_reduce_on_plateau

### PR DESCRIPTION
**Pull request by egillax**
Added a new learning rate scheduler, lr_reduce_on_plateau with tests and documentation.

Modelled it after the pytorch version. Also made tests and made sure the test cases matched with the output from pytorch.

### CODE REVIEW

**Review by dfalbel — APPROVED** [review]
Looks good to me! Thank you very much @egillax !
